### PR TITLE
Feature/add free shipping association

### PR DESCRIPTION
**Pull request by laffy**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ class Cart < FlexCommerceApi::ApiBase
     has_one :shipping_address, class_name: "::FlexCommerce::Address"
     has_one :billing_address, class_name: "::FlexCommerce::Address"
     has_one :shipping_method, class_name: "::FlexCommerce::ShippingMethod"
+    has_one :free_shipping_promotion, class_name: "::FlexCommerce::FreeShippingPromotion"
```
Comment (mufudzi-masaire) [inline app/models/cart.rb]: can't we use existing `Promotion` model? eg. - 

```ruby 
 has_one :free_shipping_promotion, class_name: "::FlexCommerce::Promotion"
```

Comment (laffy) [inline app/models/cart.rb]: It's a different object though in the API - as it's not listed with the other promotions.

File: app/models/free_shipping_promotion.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require "flex_commerce_api/api_base"
+module FlexCommerce
+  #
+  # A flex commerce discount summary model
```
Comment (garytaylor) [inline app/models/free_shipping_promotion.rb]: Please update the comment to reflect reality

File: lib/flex/commerce/api/version.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 module Flex
   module Commerce
     module Api
-      VERSION = "0.3.1"
+      VERSION = "0.3.31"
```
Comment (garytaylor) [inline lib/flex/commerce/api/version.rb]: If you are making any more changes, please remove this file - Ive been meaning to do it for ages

File: spec/fixtures/carts/singular.json [hunk 4]
```diff
@@ -44,6 +44,12 @@
           }
 
         ]
+      },
+      "free_shipping_promotion": {
+        "data": {
+          "id": "1",
+          "type": "free_shipping_promotion"
```
Comment (garytaylor) [inline spec/fixtures/carts/singular.json]: This is referring to a non existent record in the "included" objects

File: spec/fixtures/carts/singular_with_checkout.json [hunk 5]
```diff
@@ -45,6 +45,12 @@
 
         ]
       },
+      "free_shipping_promotion": {
+        "data": {
+          "id": "1",
+          "type": "free_shipping_promotion"
```
Comment (garytaylor) [inline spec/fixtures/carts/singular_with_checkout.json]: This is referring to a non existent record in the included objects

File: lib/flex/commerce/api/version.rb (hunk 3 above)
Comment (laffy) [inline lib/flex/commerce/api/version.rb]: Do you want the entire tree structure gone too, it's all empty?

File: spec/fixtures/carts/singular_with_checkout.json (hunk 5 above)
Comment (laffy) [inline spec/fixtures/carts/singular_with_checkout.json]: Already addresses this in later commit.

**Pull request merged**